src/prototype.rs

```diff
@@ -1546,51 +1594,223 @@ fn make_code_span(tree: &mut Tree<Item>, s: &str, open: usize, close: usize) {
     }
 }
 
-/// Handle inline HTML and code spans.
+fn scan_link_destination_plain(scanner: &mut InlineScanner) -> Option<String> {
```

Comment:
Wouldn't it be nicer to implement these as methods on `InlineScanner`?

Reply:
Also, do these methods replace functions in `scanners.rs`?

Reply:
They do replace functions in `scanners.rs`. In general, I haven't been aggressive in deleting old code, but a major cleanup is due soon.

Regarding whether they're methods on `InlineScanner`, that's something of a style choice, but I think of `InlineScanner` as providing access to text that might be non-contiguous because it doesn't include block structure (eg `>` for blockquote), and a function like this as consuming it.